pay-api/src/pay_api/services/base_payment_system.py

```diff
@@ -234,23 +235,61 @@ def release_payment_or_reversal(invoice: Invoice, transaction_status=Transaction
             )
 
     @staticmethod
-    def _refund_and_create_credit_memo(invoice: InvoiceModel):
+    def validate_refund_amount(refund_amount, max_amount):
+        """Validate refund amount."""
+        if refund_amount > max_amount:
+            current_app.logger.error(
+                f"Refund amount {str(refund_amount)} " f"exceeds maximum allowed amount {str(max_amount)}."
+            )
+            raise BusinessException(Error.INVALID_REQUEST)
+
+    @staticmethod
+    def _refund_and_create_credit_memo(invoice: InvoiceModel, refund_partial: List[RefundPartialLine] = None):
         # Create credit memo in CFS if the invoice status is PAID.
         # Don't do anything is the status is APPROVED.
-        current_app.logger.info(f"Creating credit memo for invoice : {invoice.id}, {invoice.invoice_status_code}")
+        is_partial = refund_partial is not None and len(refund_partial) > 0
+
+        current_app.logger.info(
+            f"Creating {'partial' if is_partial else ''} credit memo for invoice: "
```

Comment:
why not just include the amount you're refunding for the total of the invoice? no need for this: 

`{'partial' if is_partial else ''}`

Reply:
f"Creating $30 credit memo for $50 invoice : "

Reply:
reverted